### PR TITLE
Feat/route updating

### PR DESCRIPTION
**Pull request by Parle648**
## Describe your changes

- Delete order from route; when we remove order from route we:
  1. recalculate and save in DB new route distance in km
  2. rerender route distance value of FE
  3. delete order from route details page 
  4. remake route (on the map)
- Delete route; when we delete the route we:
  1. If route deleted successfully we redirect user to /routes + show toast 
  2. If route isn't exist in DB -> user will see error toast
- Change colors for status:
  1. Completed from gray to -> purple
  2. Not arrived from orange to -> brown

## Loom video
https://www.loom.com/share/2a6cb0eb4cd24d73858d9d15da081832?sid=5534f21f-5b86-4b33-9040-817bc4145a25

## Issue ticket code (and/or) and link

- [colors for statuses](https://codecraftersintership.atlassian.net/browse/SCRUM-138?atlOrigin=eyJpIjoiZTUyMTFjZDdkY2I0NDhhMzk3Y2Y5NWNiNWRjZjBiYjIiLCJwIjoiaiJ9)
- [delete order from routes story UI](https://codecraftersintership.atlassian.net/browse/SCRUM-115)
- [delete order from routes story connect Fe to Be](https://codecraftersintership.atlassian.net/browse/SCRUM-117)
- [delete order UI + connect Fe to Be task](https://codecraftersintership.atlassian.net/browse/SCRUM-141)

### **General**

- [x] Assigned myself to the PR
- [x] Assigned the appropriate labels to the PR
- [x] Assigned the appropriate reviewers to the PR
- [x] Updated the documentation
- [x] Performed a self-review of my code
- [x] Types for input and output parameters
- [x] Don't have "any" on my code
- [x] Used the try/catch pattern for error handling
- [x] Don't have magic numbers
- [x] Compare only with constants not with strings
- [x] No ternary operator inside the ternary operator
- [x] Don't have commented code
- [x] no links in the code, env links should be in env file (for example: server url), constant links (for example default avatar URL) should be in constant file.
- [x] Used camelCase for variables and functions
- [x] Date and time formats are on the constants
- [x] Functions are public only if it's used outside the class
- [x] No hardcoded values
- [x] covered by tests
- [x] Check your commit messages meet the [conventional commit format](https://www.conventionalcommits.org/en/v1.0.0/).

### Frontend

- [x] Components and business logic are separated
- [x] Colors, Font Size, and Font Name is on the theme or in the constants
- [x] No text in the components, use i18n approach
- [x] No inline styles
- [x] Imports are absolute
- [x] Attach a screenshot if PR has visual changes.

### CODE REVIEW

**Review by AlexeyBezrebriy — APPROVED** [review]

File: src/pages/RouteDetails/components/RouteDetailsControlBtns/useDeleteRoute.ts [hunk 1]
```diff
@@ -0,0 +1,34 @@
+import { setisVisible } from '@/store/slices/loaderSlice';
+import { RootState, store } from '@/store/store';
```
Comment (oyakovytskyi) [inline src/pages/RouteDetails/components/RouteDetailsControlBtns/useDeleteRoute.ts]: 1.1.14

File: src/pages/RouteDetails/components/RouteInform/components/OrderDetails/index.tsx [hunk 2]
```diff
@@ -43,8 +53,21 @@ const OrderDetails: FC<OrderDetailsProps> = ({ city, startTime }) => {
...
+        onClose={toggleIsRemoveOrder}
+        onConfirm={() => {
+          handleDelete(id);
+          toggleIsRemoveOrder();
+        }}
+        heading={`This operation delete order #${id}`}
+        mainMessage={'Do you really want to remove this order from route?'}
+        subMessage={'You will not be able to restore it'}
+        cancelText={'Close'}
+        confirmText={'Remove'}
```
Comment (oyakovytskyi) [inline src/pages/RouteDetails/components/RouteInform/components/OrderDetails/index.tsx]: 1.3.3

File: src/pages/RouteDetails/components/RouteInform/index.tsx [hunk 3]
```diff
@@ -14,6 +14,12 @@ import { driverTitleStyles, routeDateStyles, routeIdStyles } from './styles';
 const RouteInformBlock: FC<{
   routeDetails: RouteInform;
 }> = ({ routeDetails }) => {
+  console.log('render');
+
+  useEffect(() => {
+    console.log('re-render');
```
Comment (oyakovytskyi) [inline src/pages/RouteDetails/components/RouteInform/index.tsx]: 1.1.8

File: src/pages/RouteDetails/useGetRoute.ts [hunk 4]
```diff
@@ -6,14 +6,20 @@ import { t } from 'i18next';
 import { RouteInform } from '@/interfaces/interfaces';
 import { START_ROUTE_POINT } from '@/constants/constants';
 import { getRouteDetails } from './components/RouteInform/api/getRouteDetails';
+import { useSelector } from 'react-redux';
+import { RootState, store } from '@/store/store';
+import axios from 'axios';
```
Comment (oyakovytskyi) [inline src/pages/RouteDetails/useGetRoute.ts]: 1.1.14

File: src/pages/RouteDetails/components/RouteDetailsControlBtns/useDeleteRoute.ts (hunk 1 above)
Comment (Parle648) [reply src/pages/RouteDetails/components/RouteDetailsControlBtns/useDeleteRoute.ts]: forget to push changes, done

File: src/pages/RouteDetails/components/RouteInform/components/OrderDetails/index.tsx (hunk 2 above)
Comment (Parle648) [reply src/pages/RouteDetails/components/RouteInform/components/OrderDetails/index.tsx]: done

File: src/pages/RouteDetails/components/RouteInform/index.tsx (hunk 3 above)
Comment (Parle648) [reply src/pages/RouteDetails/components/RouteInform/index.tsx]: same

File: src/pages/RouteDetails/useGetRoute.ts (hunk 4 above)
Comment (Parle648) [reply src/pages/RouteDetails/useGetRoute.ts]: same

**Review by oyakovytskyi — APPROVED** [review]

**Pull request merged**